test/e2e/prometheus_test.go

```diff
@@ -1553,6 +1553,54 @@ func testOperatorNSScope(t *testing.T) {
 			t.Fatalf("expected alert %q not to fire", secondAlertName)
 		}
 	})
+
+	t.Run("PromInstanceMultiNS", func(t *testing.T) {
+		ctx := framework.NewTestCtx(t)
+		defer ctx.Cleanup(t)
+
+		operatorNS := ctx.CreateNamespace(t, framework.KubeClient)
+		prometheusNS := ctx.CreateNamespace(t, framework.KubeClient)
+		ruleNS := ctx.CreateNamespace(t, framework.KubeClient)
+		arbitraryNS := ctx.CreateNamespace(t, framework.KubeClient)
+
+		ctx.SetupPrometheusRBAC(t, prometheusNS, framework.KubeClient)
+
+		prometheusNamespaceSelector := map[string]string{"prometheus": prometheusNS}
+
+		for _, ns := range []string{ruleNS, arbitraryNS} {
+			err := testFramework.AddLabelsToNamespace(framework.KubeClient, ns, prometheusNamespaceSelector)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		// Prometheus Operator only watches prometheusNS and ruleNS, not arbitraryNS.
+		// this is the crux of the test: we configure to watch ruleNS for Rules and promNS for Prom CRs
+		// and create Prom CRs in both.
+		// then we test that Prometheus instance is created in promNS and NOT created with an error in ruleNS
+		err := framework.CreatePrometheusOperator(operatorNS, *opImage, []string{ruleNS}, []string{prometheusNS})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		p := framework.MakeBasicPrometheus(prometheusNS, name, name, 1)
+		p.Spec.EvaluationInterval = "1s"
+		p, err = framework.CreatePrometheusAndWaitUntilReady(prometheusNS, p)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		pMissed := framework.MakeBasicPrometheus(ruleNS, name, name, 1)
+		pMissed.Spec.EvaluationInterval = "1s"
+		pMissed, err = framework.CreatePrometheusAndWaitUntilReady(ruleNS, pMissed)
+		if err == nil {
```

Comment:
I am not sure how to test that it ignores Prometheus CR in a given namespace. I didn't come up with anything better than checking that error (timeout) happens indeed, but it makes test run longer :(